test/unit/components/VDataTable/VDataTable.spec.js

```diff
@@ -339,10 +339,10 @@ test('VDataTable.vue', ({ mount, compileToFunctions }) => {
 
   it('should render correct colspan when using headers-length prop', async () => {
     const data = dataTableTestData()
-    data.propsData.headersLength = 11
+    data.propsData.headersLength = 3
```

Comment:
If 11 was failing then there's probably something wrong.

Reply:
need to add 11 items in the `headers` array if we insist to use 11, because the program depends on the computed properties `headerColumns` for width calculation

ps. Feel like the value of `headersLength` should align with the length of the headers array

```
  function dataTableTestData () {
    return {
      propsData: {
        pagination: {
          descending: false,
          sortBy: 'col1',
          rowsPerPage: 1,
          page: 1
        },
        headers: [
          { text: 'First Column', value: 'col1', class: 'a-string' },
          { text: 'Second Column', value: 'col2', sortable: false },
          { text: 'Third Column', value: 'col3', class: ['an', 'array'] }
        ],
        items: [
          { other: 1, col1: 'foo', col2: 'a', col3: 1 },
          { other: 2, col1: null, col2: 'b', col3: 2 },
          { other: 3, col1: undefined, col2: 'c', col3: 3 }
        ]
      }
    }
  }
```